src/frontend/query_protocol.ml

```diff
@@ -142,8 +142,7 @@ type _ t =
        ] t
   | Syntax_document
     : Msource.position
-    ->[ `Found of syntax_doc_result
-      | `No_documentation
+    ->[ `Found of syntax_doc_result option
```

Comment:
I don't understand why you made that change ? The previous type was perfectly fine.
`Found None` reads like a contradiction...

Reply:
making the return value `syntax_result option`  allows us to have more freedom in other implementations like `query_json` and `ocaml-lsp`. 
but looking at it now I agree it does look weird . I'll refactor it so it has a more meaningful structure.

Reply:
I don't see why it would give us more freedom in`query_json` or `ocaml-lsp` since we encode the exact same amount of information: we could simply return `syntax_doc_result option` since there is only two cases. Either that or custom variant like `` `Found`` and `` `No_documentation`. All these options encode the same things and allow for the same flexibility. We often encourage more verbose names instead of plain `option`, especially when writing protocols, since it allows future readers to understand better the purpose just by reading the types.

Reply:
ah I see, I was going for simply returning `syntax_doc_result option` but I like the importance of using verbose names as you mentioned above. I think it makes more sence to use `Found` and `No_documentation`